### ISSUE TITLE
[gpx import] No import of comments / description of waypoints

### ISSUE THREAD

**Issue by GerdBlomb** [organicmaps/organicmaps#5643]
**Describe the issue**
I want to do my travel plannings with Organic Maps in the future. I like the clear UX/UI and the functions. Some functions are still missing, but it works so far better for me than OSMand. 

Unfortunately my descriptions in the waypoints are not imported, when using gpx files. I create my files on Windows with a desktop app (Quo Vadis and RouteConverter) 

The ```<cmt>...</cmt>``` tag seams to be ignored.

Please let me know if i can assist here. Maybe with testing...

Thanks for all the work you put into it. It's an awesome project.

I know that i can workaround that with using kml/kmz files. 

**Steps to reproduce**
1. Import the attached GPX file
2. After import go the new list and examine the waypoint in that list
3. The "lorem" text description is not there
4. The import in other gpx capable apps works fine

**Expected behavior**
Import all gpx tags from gpx 1.1 specs

**Screenshots**
none

**System information:**
 - Operating system and its version: iOS 16.5
 - Organic Maps version: 202307.13-3 (beta and latest stable tested)
 - Device Model: iPad Pro Gen. 3

**Additional context**
s. attached file
[test.zip](https://github.com/organicmaps/organicmaps/files/12257583/test.zip)

**Comment by cyber-toad** [comment]
> @cyber-toad can you please take a look? Is cmt an official gpx tag?

@biodranik yes, it is in spec.

[<cmt> GPS comment of the waypoint](https://www.topografix.com/gpx_manual.asp#cmt)
[<desc> Descriptive description of the waypoint](https://www.topografix.com/gpx_manual.asp#desc)

Looks like something close to description, that we parse already. It is easy to extract <cmt> value, but where do we want to put it?
In KML we have

```
      else if (prevTag == "mwm:annotation" && m_attrCode >= 0)
        m_categoryData->m_annotation[m_attrCode] = value;
```
do we want to use annotation field? Or something like merge with desc and put to description?

**Comment by biodranik** [comment]
@GerdBlomb according to the spec, user-facing comments should be stored in `<desc>`, not in `<cmt>`, which is intended to send the short comment to the GPS device.

The good question is why Quo Vadis and RouteConverter use that tag instead of `<desc>`, and can they use desc instead or not?

@cyber-toad isn't annotation used for list' descriptions? Here we're talking about each bookmark/track's comment.

Let's append it smartly to an existing description (you may write a few unit tests for these cases).

**Comment by cyber-toad** [comment]
> @cyber-toad isn't annotation used for list' descriptions? Here we're talking about each bookmark/track's comment.
> 
> Let's append it smartly to an existing description (you may write a few unit tests for these cases).

@biodranik Yep, my mistake - I looked into wrong structure. It seems that it was requested for other applications also: https://help.locusmap.eu/topic/gpx-extension-supported, so I'll add it. 
The only thing that I worry about is that is it possible that it will have some tech details and it will pollute description field if it already has some meaningful data? Maybe use cmt only if desc is missing? 

@GerdBlomb do you have both desc and cmt tags in tracks provided by your app?

**Comment by biodranik** [comment]
@cyber-toad if the description is empty, then it's clear: just use cmt. If description is not empty, then append cmt at the end via `\n\n` delimiter (considering that there is no `\n` at the end of the description already).

I assume that the parser can see cmt first, before the desc too. Then the same logic above should be used to append desc to cmt.

**Comment by GerdBlomb** [comment]
> @GerdBlomb do you have both desc and cmt tags in tracks provided by your app (not a case for your sample, but what about original files)?

No just the ```cmt``` Tags is in there. The example file is complete. And yes, i think you are right: ```desc``` should be the tag to be used. Until today i was not ware of that, because other tools in my portfolio do handle the ```cmt``` tags (as well). I've never missed the text in any other app.

I find this could be the smart solution: If it is there append it to the description (s. above).

Thanks for your investigation on this one!

**Comment by GerdBlomb** [comment]
I've mad a short test. I added a ```cmt``` and a ```desc``` tag to the test gpx file and imported it into 

**Quo Vadis**
Both text parts are imported with a paragraph line break:

![quovadis](https://github.com/organicmaps/organicmaps/assets/80357008/1e4bcae4-d1cf-4f1a-ad37-81754dad08c4)


**QMapShack**
Both text are imported with a header (Comment and description):

![qmapshack](https://github.com/organicmaps/organicmaps/assets/80357008/2ace4ac8-aa94-47a8-ac9f-c01dcd7256fb)

New test file
[test_file_cmt_desc.zip](https://github.com/organicmaps/organicmaps/files/12273016/test_file_cmt_desc.zip)

**Comment by GerdBlomb** [comment]
I did another test:

I've created a new waypoint in QuoVadis and created also a new waypoint in QMapShack. After that i made a  gpx export in each application.

QuoVadis does not support a Description. Id just uses ```cmt```. 

QMapShack makes a difference between the two tags and one can enter text for each tag / field. This results in a file which contains both tags.

I've attached the export results in the Zip file below. They are original as the apps wrote them out. No modifications habe been made after the exports.

[two_export_examples.zip](https://github.com/organicmaps/organicmaps/files/12273292/two_export_examples.zip)
